src/deployment.ts

```diff
@@ -215,7 +216,7 @@ export class Deployment extends workload.Workload {
    */
   public _toKube(): k8s.DeploymentSpec {
     return {
-      replicas: this.replicas,
+      replicas: this.replicas ?? 2,
```

Comment:
Not sure I follow this, won't this interfere with HPA if configured on this deployment? We spoke about providing a default replica count **only if** an HPA isn't defined, but this doesn't seem to consider it.

Reply:
Yeah something needs to change here. When writing this I was under the impression that `_toKube()` would only be called if it wasn't associated with an HPA since we now have `toScalingTarget()`. I didn't consider that `_toKube()` can still be called by the chart/other associated constructs. I'll have to think about this.